pkg/generic/httppbthrift_codec.go

```diff
@@ -114,7 +123,23 @@ func (c *httpPbThriftCodec) getMessageReaderWriter() interface{} {
 		return errors.New("get parser PbServiceDescriptor failed")
 	}
 
-	return thrift.NewHTTPPbReaderWriter(svcDsc, pbSvcDsc)
+	cachedData := c.cachedReaderWriterData.Load()
+	if cachedData != nil {
+		if cd, ok := cachedData.(*cachedHTTPPbReaderWriterData); ok && cd.svcDsc == svcDsc {
```

Comment:
Why don't update readerWriter when updating svcDsc, in this way you can just get the rw you want, and no need to do this judgement and updating.

Reply:
I actually thought of that, but I decided not to do that somehow.
Let me remember (I also feel just updating in update() is enough now)

Reply:
I remembered, and now I think you're right. I'll modify.

Reply:
@jayantxie Can't it lead to data race? messageReaderWriter is not only related to xxxcode, but also related to message writer and reader, so I wanted to protect by locking when updating readerWriter

Reply:
Where is `UpdateIDL` logic? I didn't figure out

Reply:
@AsterDY It's in thriftidl_provider.go

Reply:
Updated the logic